.github/workflows/ci.yml

```diff
@@ -0,0 +1,482 @@
+name: CI
+
+on:
+  [push, pull_request]
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  RELEASE: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags/test-v') }}
+  # The test-installer job can use an old existing image rather than the most
+  # recently built image; the 'needs' can be commented out to allow the
+  # test-installer job to run independently (and therefore faster)
+  INSTALLER_USE_OLD_IMAGE: false
+  INSTALLER_OLD_RUNID: 3204825457
+  # The test-installer job can run with the pyinstaller debug bundle
+  INSTALLER_USE_DEBUG: false
+
+jobs:
+  matrix:
+    name: Generate test matrix
+
+    #if: false    # uncomment to prevent the rest of the workflow running
+
+    runs-on: ubuntu-latest
+
+    outputs:
+      matrix-json: ${{ steps.set-matrix.outputs.matrix }}
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - uses: actions/checkout@v3
+      - id: set-matrix
+        run: python .github/workflows/matrix.py >> $GITHUB_OUTPUT
+
+  build-matrix:
+    needs: [matrix]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.matrix.outputs.matrix-json) }}
+
+    name: "${{ matrix.job_name }}"
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Configuration of this job
+        run: |
+          echo '${{ toJson(matrix) }}'
+
+      - name: Obtain SasView source from git
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: |
+            **/ci.yml
+            **/requirements*.txt
+
+      ### Installation of build-dependencies
+
+      - name: Install X11 libraries (Linux)
+        if: ${{ startsWith(matrix.os, 'ubuntu') }}
+        run: |
+          sudo apt install libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils xvfb
+
+      - name: Install lxml and MPL (Windows + Linux)
+        if: ${{ !startsWith(matrix.os, 'macos') }}
+        run: |
+          python -m pip install matplotlib~=3.5.2
+          python -m pip install lxml
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install wheel setuptools
+
+          python -m pip install numpy scipy==1.7.3 pytest pytest-mock pytest_qt
+          python -m pip install tinycc h5py pyparsing html5lib reportlab==3.6.6 pybind11 appdirs
+          python -m pip install six numba mako ipython qtconsole xhtml2pdf pylint
+          python -m pip install qt5reactor periodictable uncertainties dominate importlib_resources
+          python -m pip install html2text
```

Comment:
Perhaps out of scope but reproducible builds should pin package version numbers. Because we do significant manual testing prior to release we probably don't want packages changing out from under us. Automated testing should happen against the pinned versions and against the latest versions so we don't get too out of date.

You could do this with a pair of `requirements.txt`. These even allow os-dependent installs ([stackoverflow link](https://stackoverflow.com/questions/29222269/is-there-a-way-to-have-a-conditional-requirements-txt-file-for-my-python-applica/35614580#35614580)).

Reply:
Yes, that particular can of worms is out of scope for this. The small amount of pinning that is already here has pin conflicts that need to be forced/ignored.